services/core/shared/core/compat/sdk_v5/coreApi.js

```diff
@@ -61,25 +76,26 @@ const getBlockByID = async id => {
 
 const getBlocksByIDs = async ids => {
 	try {
+		// File based Genesis block handling
+		const genesisBlockId = getGenesisBlockId();
+		if (ids.includes(genesisBlockId)) {
+			const remainingIds = ids.filter(id => id !== genesisBlockId);
+			const genesisBlockResult = await getBlockByID(genesisBlockId);
+			if (remainingIds.length) {
+				const { data: [genesisBlock] } = genesisBlockResult;
+				const { data: [...remainingBlocks] } = await getBlocksByIDs(remainingIds);
+				return { data: [genesisBlock, ...remainingBlocks] };
+			}
+			return genesisBlockResult;
```

Comment:
Please avoid code duplication. The retrieval logic gets complex and all functions could call data with multiple block methods, either by IDs or range.

Reply:
`getBlocksByIDs` and `getBlocksByHeightBetween` utilise a similar approach but the implementation invoke their corresponding singular methods i.e. `getBlockByID` and `getBlocksByHeightBetween`.

Reply:
It will be updated in #642